### PR TITLE
QuantileRegression

### PR DESCRIPTION
**Pull request by Garve**
Alright, second try.

### CODE REVIEW

File: tests/test_estimators/test_lad_regression.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def test_coefs_and_intercept__no_noise_regularization(coefs, intercept):
     """Test model with regularization. The size of the coef vector should shrink the larger alpha gets."""
     X, y = _create_dataset(coefs, intercept)
 
-    lads = [LADRegression(alpha=alpha, l1_ratio=0.5).fit(X, y) for alpha in range(3)]
+    lads = [LADRegression(alpha=alpha, l1_ratio=0.).fit(X, y) for alpha in range(3)]
```
Comment (koaning) [inline tests/test_estimators/test_lad_regression.py:62]: Is there a reason why the LADRegression test needed to be changed?

**Review by koaning — CHANGES_REQUESTED** [review]
Only did a quick review now, left a comment. Hope to dive in more this weekend.

File: tests/test_estimators/test_lad_regression.py (hunk 1 above)
Comment (Garve) [reply tests/test_estimators/test_lad_regression.py:62]: I just noticed by accident. Dunno why I put a 0.5 there in the first place.

I just wanted to impose an L2 regularization, and that's what I check I also check afterward. in the next lines, I check if the sum of squares of coefficients goes down the larger alpha gets, and not a mix of the L1 and L2 norms of the coefficient.

The test still worked for the LADRegression, but it was just a happy little accident. For the QuantileRegression, the same check didn't pass, and that's when I noticed this mistake.

File: tests/test_estimators/test_quantile_regression.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+@pytest.mark.parametrize("coefs, intercept", test_batch)
+@pytest.mark.parametrize("quantile", np.linspace(0, 1, 7))
+def test_quantile(coefs, intercept, quantile):
+    """Tests with noise on an easy problem. A good score should be possible."""
+    X, y = _create_dataset(coefs, intercept, noise=1.0)
+    quant = QuantileRegression(quantile=quantile)
+    quant.fit(X, y)
+
+    np.testing.assert_almost_equal((quant.predict(X)>=y).mean(), quantile, decimal=2)
```
Comment (koaning) [inline tests/test_estimators/test_quantile_regression.py:54]: I didn't have the time to download and go fully into the numerics myself, but from my end ... the fact that this test passes gives me enough confidence that it's green to merge. 

Before I do that. @MBrouns just to check if you had any comments?

**Review by koaning — COMMENTED** [review]
Green on my end, just want to give @MBrouns the opportunity to pitch in.

**Comment by MBrouns** [comment]
I'm not sure whether I love the decorators for the regularization terms. What's wrong with doing

```
def grad_quantile_loss(params):
      return (
          -(
              sample_weight
              * np.where(X @ params < y, self.quantile, 1 - self.quantile)
              * np.sign(y - X @ params)
          )
          @ X
          / X.shape[0]
         + self._grad_loss_regularize(params)
      )
```

**Comment by Garve** [comment]
Hi!

This should be fine, too. But then we should change it for all regressors, perhaps in a different PR.

**Comment by koaning** [comment]
@Garve from my end it's preferable to do that in this PR. Once this PR is in, it's time for another release.

**Comment by koaning** [comment]
Oh! And there is a minor thing actually! Could you double check that the component list in the `README.md` is up to date?

**Comment by Garve** [comment]
Decorators gone, readme is up to date.

**Comment by koaning** [comment]
LGTM

**Review by koaning — APPROVED** [review]

**Pull request merged**